### ISSUE TITLE
UI: Dropdown doesn't list all the entries

### ISSUE THREAD

**Issue by vishesh92** [apache/cloudstack#9729]
##### ISSUE TYPE
 * Enhancement Request

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
All versions which have the new UI
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

The dropdown in the UI fetches all domains at once with the default page size. If the actual count of resources is higher, the keyword search doesn't work and the domains after the default page size can't be selected in the UI.

##### STEPS TO REPRODUCE

1. Create 10 domains with different names.
2. Set the default page size to 5. 
3. The domain list while launching or importing a VM, doesn't show all the records.

~~~

~~~

##### EXPECTED RESULTS

~~~
I should be able to see all records or search by keyword and select. Or allow entering a UUID manually.
~~~

##### ACTUAL RESULTS

~~~
All entries in the drop down are not listed.
~~~

**Comment by DaanHoogland** [comment]
Hey @vishesh92 have you verified that this problem is not in the API? It sounds like it could have to do with the pagesize and page parameters.

**Comment by vishesh92** [comment]
> Hey @vishesh92 have you verified that this problem is not in the API? It sounds like it could have to do with the pagesize and page parameters.

The problem could be fixed by increasing the pagesize parameter but that would also slow down the UI. And if the number of entries are let's say 10000, it can also increase the load on the database.

**Comment by DaanHoogland** [comment]
> > Hey @vishesh92 have you verified that this problem is not in the API? It sounds like it could have to do with the pagesize and page parameters.
> 
> The problem could be fixed by increasing the pagesize parameter but that would also slow down the UI. And if the number of entries are let's say 10000, it can also increase the load on the database.

What you are saying is that the problem is not in the UI, right?

**Comment by DaanHoogland** [comment]
> This is an enhancement request. As of now, the end user will have to use the API instead of the UI.
> 
> UI should be making search queries with keyword when text is entered in the drop down to ensure that the user can search and select the desired resource irrespective of how many entries are present.
> 
> Another possible fix is to allow the user to enter the UUID even if it's not present in the drop down.

Yes, but still it is the API that must be fixed, unless you say that giving the page and pagesize parameters on the CLI will yield the requested results.

**Comment by Imvedansh** [comment]
@vishesh92 Could you please provide more details about the issue so I can  investigate it further?
 Here's what I've observed so far: 
when I set the page size to 5, I was able to search for domains, and although only 5 domains were displayed in the list, I could still search for all domains by their names.
However, with the page size set to 5, other UI elements became unresponsive, and only the domain section remained accessible.

Could you elaborate on the issue or share additional context? I will work on this.

**Comment by vishesh92** [comment]
> @vishesh92 Could you please provide more details about the issue so I can  investigate it further?
>  Here's what I've observed so far: 
> when I set the page size to 5, I was able to search for domains, and although only 5 domains were displayed in the list, I could still search for all domains by their names.
> However, with the page size set to 5, other UI elements became unresponsive, and only the domain section remained accessible.
> 
> Could you elaborate on the issue or share additional context? I will work on this.

@Imvedansh the issue is not with search but at places where we need to select a resource using a type ahead input.

**Comment by sureshanaparti** [comment]
This is partially addressed here: https://github.com/apache/cloudstack/pull/11991. Still some drop downs in Create Volume, Register Template/ISO from URL, Upload Template/ISO from Local, etc to be addressed. cc @DaanHoogland @abh1sar @vishesh92

**Comment by DaanHoogland** [comment]
@sureshanaparti (cc @vishesh92 ) any idea if, or how complete this list is?

**Comment by vishesh92** [comment]
> [@sureshanaparti](https://github.com/sureshanaparti) (cc [@vishesh92](https://github.com/vishesh92) ) any idea if, or how complete this list is?

There are a lot of other places which needs this change and will require searching through the code to update this. I don't have a complete list as of now. But I doubt the PR #11991 covers even 25% of the usage.

**Comment by DaanHoogland** [comment]
so should we have the ambition to solve this before 20.3 then?

**Comment by DaanHoogland** [comment]
@vishesh92 @sureshanaparti moving this oput of planned milestones. please escalate if you feel it deserves mote prio.

**Comment by DaanHoogland** [comment]
@vishesh92 , @sureshanaparti , care to give this another go?
